### PR TITLE
Insert study in directory server 

### PR DESCRIPTION
**Pull request by ChamseddineBhd**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
     private final StudyService studyService;
 
+    public static final String TMP_LEGACY_DIRECTORY = "11111111-2222-3333-4444-555555555555";
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyController.java:36]: What is this ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 2]
```diff
@@ -246,37 +249,93 @@ private static CreatedStudyBasicInfos toCreatedStudyBasicInfos(StudyEntity entit
...
-                                    description, t.getT2(), caseUuid, false, LoadFlowStatus.NOT_DONE, null, toEntity(loadFlowParameters), null);
-                        })
-                        .subscribeOn(Schedulers.boundedElastic())
-                        .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
-                        .doFinally(r -> deleteStudyIfNotCreationInProgress(s.getStudyUuid(), userId).subscribe())
-                        .subscribe()
-                );
+    private Mono<Void> insertDirectoryElement(UUID parentDirectoryUuid, DirectoryElement directoryElement) {
+        //To be removed after removing the study manager
+        if (parentDirectoryUuid == null) {
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyService.java:254]: How can it be null ? you give a default value in the controller

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 3]
```diff
@@ -246,37 +249,93 @@ private static CreatedStudyBasicInfos toCreatedStudyBasicInfos(StudyEntity entit
...
-                                    LoadFlowParameters loadFlowParameters = LoadFlowParameters.load();
-                                    return insertStudy(s.getStudyUuid(), studyName, userId, isPrivate, t.getT1().getNetworkUuid(), t.getT1().getNetworkId(),
-                                            description, t.getT2(), uuid, true, LoadFlowStatus.NOT_DONE, null, toEntity(loadFlowParameters), null);
-                                }))
-                        .subscribeOn(Schedulers.boundedElastic())
-                        .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
-                        .doFinally(r -> deleteStudyIfNotCreationInProgress(s.getStudyUuid(), userId).subscribe())  // delete the study if the creation has been canceled
-                        .subscribe()
-                );
+                .doOnSuccess(s -> insertDirectoryElement(parentDirectoryUuid, s) // insert study in directory server
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyService.java:303]: What if insertDirectoryElement fails or throw exception ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 4]
```diff
@@ -246,37 +249,93 @@ private static CreatedStudyBasicInfos toCreatedStudyBasicInfos(StudyEntity entit
...
+                                            deleteDirectoryElement(s.getStudyUuid()).then(deleteStudyIfNotCreationInProgress(s.getStudyUuid(), userId)).subscribe();
+                                        })
+                                        .doOnSuccess(r -> deleteStudyIfNotCreationInProgress(s.getStudyUuid(), userId).subscribe())
+                        ).subscribe());
+    }
+
+    public Mono<BasicStudyInfos> createStudy(String studyName, Mono<FilePart> caseFile, String description, String userId, Boolean isPrivate, UUID parentDirectoryUuid) {
+        return insertStudyCreationRequest(studyName, userId, isPrivate)
+                .map(StudyService::toBasicStudyInfos)
+                .doOnSuccess(s -> insertDirectoryElement(parentDirectoryUuid, s)  // insert study in directory server
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyService.java:323]: same, what if insertDirectoryElement fails ?

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 5]
```diff
@@ -395,7 +454,7 @@ public StudyEntity doGetStudyWithPreFetchedLoadFlowResultAndUpdateIsPrivate(UUID
 
     private Mono<StudyCreationRequestEntity> insertStudyCreationRequest(String studyName, String userId, boolean isPrivate) {
         return insertStudyCreationRequestEntity(studyName, userId, isPrivate)
-                .doOnSuccess(s -> emitStudiesChanged(s.getId(), userId, isPrivate));
+                .doOnSuccess(ignore -> emitStudiesChanged(ignore.getId(), userId, isPrivate));
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/StudyService.java:457]: why this rename ?

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 6]
```diff
@@ -221,6 +224,8 @@ public MockResponse dispatch(RecordedRequest request) throws InterruptedExceptio
                             .build(), "sa.stopped");
                     return new MockResponse().setResponseCode(200)
                             .addHeader("Content-Type", "application/json; charset=utf-8");
+                } else if (path.matches("/v1/directories/.*")) {
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:227]: it's better if you put the exact route. Here any calls to directory server will return 200 so you don't test when the call fail. And it'll be easier for anyone to add new tests without having to touch your tests.

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 7]
```diff
@@ -1339,6 +1344,8 @@ private void createStudy(String userId, String studyName, UUID caseUuid, String
         assertNotEquals(isPrivate, headers.get(HEADER_IS_PUBLIC_STUDY));
         assertEquals(UPDATE_TYPE_STUDIES, headers.get(HEADER_UPDATE_TYPE));
 
+        message = output.receive(1000);
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1347]: Add a comment to tell what this receive is supposed to receive

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 8]
```diff
@@ -1400,6 +1408,8 @@ private void createStudy(String userId, String studyName, String fileName, Strin
         assertNotEquals(isPrivate, headers.get(HEADER_IS_PUBLIC_STUDY));
         assertEquals(UPDATE_TYPE_STUDIES, headers.get(HEADER_UPDATE_TYPE));
 
+        output.receive(1000);
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1411]: same

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 9]
```diff
@@ -1412,7 +1422,7 @@ private void createStudy(String userId, String studyName, String fileName, Strin
         assertEquals(errorMessage.length != 0 ? errorMessage[0] : null, headers.get(HEADER_ERROR));
 
         // assert that the broker message has been sent a study creation request message for deletion
-        message = output.receive(1000);
+        message = output.receive(20000);
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1425]: is this still needed ?

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/study/server/StudyController.java:36]: to make the code backward compatible with study manager. so when we create a study from the study manager we could see it in the gridexplore (legacy directory). this is an intermediate step. we will remove this when the study manager will be removed.

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 2 above)
Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/study/server/StudyService.java:254]: alright

**Review by AbdelHedhili — APPROVED** [review]

**Pull request merged**